src/activation.rs

```diff
@@ -225,6 +249,16 @@ impl IsType for RawFd {
         }
     }
 
+    fn is_stream(&self) -> bool {
+        let sock_type = getsockopt(*self, SockType).expect("Getsockopt");
```

Comment:
I think this (and the one below) on error should default to `false` instead of panic-ing.